modules/installing-gcp-user-defined-tags.adoc

```diff
@@ -0,0 +1,133 @@
+
+
+User-defined and {product-title} specific tags are applied only to resources created by {product-title} installer and its core operators such as `machine-api-provider-gcp`, `cluster-image-registry-operator`, `gcp-filestore-csi-driver-operator`, g`cp-pd-csi-driver-operator` and not on the resources created by any other operators or the kubernetes in-tree components that creates resources for example the ingress load balancers.
+
+
+User-defined tags are attached to resources created by {product-title} installer and its core operators such as `cluster-image-registry-operator` and not on the resources created by any other operators or the kubernetes in-tree components. In {product-title} version {product-version}, resources created by `machine-api-provider-gcp`, `gcp-filestore-csi-driver-operator`, `gcp-pd-csi-driver-operator` are not tagged.
+
+
+[NOTE]
+====
+Labels & tags once defined during the install-config creation phase cannot be modified or updated with new labels & tags after cluster creation.
+====
+
+
+Support for user-defined labels & tags for GCP resources is available in {product-title} version {product-version} as Technology Preview (TP).
+
+
+Service Account used for cluster creation must have the `TagUser` role assigned to it, required for tags defined at both organization and project level.
+
+
+You can use `.platform.gcp.userLabels` and `platform.gcp.userTags` fields to define the list of desired labels & tags in `install-config.yaml`. `.platform.gcp.userLabels` and `.platform.gcp.userTags` fields are a yaml array of objects type.
+
+
+.Sample `install-config.yaml` file
+[source,yaml]
+----
+apiVersion: v1
+featureSet: TechPreviewNoUpgrade
+platform:
+ gcp:
+   userLabels:
+   - key: <label_key>
+     value: <label_value>
+   userTags:
+   - parentID: <OrganizationID/ProjectID>
+     key: <tag_key_short_name>
+     value: <tag_value_short_name>
+----
+
+The user-defined labels have the following requirements:
+
+* A label key and value must have a minimum of 1 character and can have a maximum of 63 characters.
+* A label key and value must contain only lowercase letters, numeric characters, underscores, and dashes.
+* A label key must start with a lowercase letter.
+* Each resource can have multiple labels, up to a maximum of 32.
+* You can configure a maximum of 64 labels per resource.
+* The number of user-defined labels are restricted to 32 and reserved to 32 for {product-title} internal use.
+
+The user-defined tags have the following requirements:
+
+* Tag Keys and Tag Values must already exist and will not be created by {product-title}.
+* A tag `parentID` can be either `OrganizationID` or `ProjectID`.
+* An `OrganizationID` must consist of decimal numbers, and cannot have leading zeros.
+* A `ProjectID` must be 6 to 30 characters in length, can only contain lowercase letters, numbers, and hyphens, and must start with a letter, and cannot end with a hyphen.
```

Comment:
Please make above two as two subpoints of `A tag `parentID` can be either `OrganizationID` or `ProjectID`.`, if it's not already.

Reply:
ack.